packages/extension-chakra-storefront/src/setup-app.tsx

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+// Third-Party
+import React from 'react'
+import {RouteProps} from 'react-router-dom'
+
+// Platform Imports
+import {ApplicationExtension} from '@salesforce/pwa-kit-extension-sdk/react'
+import {applyHOCs} from '@salesforce/pwa-kit-extension-sdk/react/utils'
+
+// Local Imports
+import {Config} from './types'
+import {configureRoutes} from './utils/routes-utils'
+import {withChakraUI} from './components/with-chakra-ui'
+import {withCommerceSdkReact} from './components/with-commerce-sdk-react'
+import {withCurrency} from './components/with-currency'
+import {withLayout} from './components/with-layout'
+import {withMultiSite} from './components/with-multi-site'
+import {withReactIntl} from './components/with-react-intl'
+import {withStorefrontPreview} from './components/with-storefront-preview'
+
+// Pages
+import * as Pages from './pages'
+
+class RetailReactApp extends ApplicationExtension<Config> {
+    extendApp<T>(App: React.ComponentType<T>): React.ComponentType<T> {
+        // NOTE: The order of these HOCs is important!
+        const requiredHOCs = [
+            withLayout,
+            withChakraUI,
+            withCurrency,
+            withReactIntl,
+            withMultiSite,
+            withStorefrontPreview,
+            withCommerceSdkReact
+        ]
+
+        return applyHOCs(App, requiredHOCs)
+    }
+
+    extendRoutes(routes: RouteProps[]): RouteProps[] {
+        const config = this.getConfig()
+        const extensionRoutes = [
+            {
+                path: '/',
+                component: Pages.Home,
+                exact: true
+            },
+            {
+                path: '/login',
+                component: Pages.Login,
+                exact: true
+            },
+            {
+                path: '/registration',
+                component: Pages.Registration,
+                exact: true
+            },
+            {
+                path: '/reset-password',
+                component: Pages.ResetPassword,
+                exact: true
+            },
+            {
+                path: '/account',
+                component: Pages.Account
+            },
+            {
+                path: '/checkout',
+                component: Pages.Checkout,
+                exact: true
+            },
```

Comment:
An unexpected error happens when placing the order and then the checkout is stuck with the Skeletons.

<img width="1728" alt="Screenshot 2024-11-13 at 9 38 22 AM" src="https://github.com/user-attachments/assets/37a20e0e-285b-485b-a030-2a8de84e15f7">

Reply:
Looks like this was either an intermittent issue, so it was resolved with an unrelated change. 

<img width="1281" alt="image" src="https://github.com/user-attachments/assets/10373896-c51b-4b2b-8f38-a4c69036c4ce">